js/customize-snapshots.js

```diff
@@ -139,1090 +156,104 @@
-			changesetStatus = api.state( 'changesetStatus' ).get();
-			statusButton = {};
-
-			selectedStatus = changesetStatus && 'auto-draft' !== changesetStatus ? changesetStatus : 'publish';
-
-			statusButton.value = new api.Value( selectedStatus );
-			statusButton.disbleButton = new api.Value();
-			statusButton.disableSelect = new api.Value();
-			statusButton.buttonText = new api.Value();
-			statusButton.needConfirm = false;
-
-			statusButton.container = $( $.trim( wp.template( 'snapshot-status-button' )({
-				selected: selectedStatus
-			}) ) );
-			statusButton.button = statusButton.container.find( '.snapshot-status-button-overlay' );
-			statusButton.select = statusButton.container.find( 'select' );
-			statusButton.select.selectmenu({
-				width: 'auto',
-				icons: {
-					button: 'dashicons dashicons-arrow-down'
-				},
-				change: function( event, ui ) {
-					statusButton.value.set( ui.item.value );
 				},
-				select: function() {
-					if ( statusButton.hiddenButton ) {
-						statusButton.hiddenButton.text( statusButton.buttonText.get() );
-					}
-				}
-			});
-
-			selectMenuButton = statusButton.container.find( '.ui-selectmenu-button' );
-			statusButton.hiddenButton = selectMenuButton.find( '.ui-selectmenu-text' );
-			statusButton.hiddenButton.addClass( 'button button-primary' );
-
-			statusButton.dropDown = selectMenuButton.find( '.ui-icon' );
-			statusButton.dropDown.addClass( 'button button-primary' );
-
-			statusButton.updateButtonText = function( dataAttr ) {
-				buttonText = statusButton.button.data( dataAttr );
-				statusButton.button.text( buttonText );
-				statusButton.hiddenButton.text( buttonText );
-				statusButton.buttonText.set( buttonText );
-			};
-
-			statusButton.value.bind( function( status ) {
-				selectedOption = statusButton.select.find( 'option:selected' );
-				statusButton.button.data( 'alt-text', selectedOption.data( 'alt-text' ) );
-				statusButton.button.data( 'button-text', selectedOption.text() );
-				statusButton.updateButtonText( 'button-text' );
-
-				if ( 'publish' === status ) {
-					snapshot.snapshotExpandButton.hide();
-					statusButton.button.data( 'confirm-text', selectedOption.data( 'confirm-text' ) );
-					statusButton.button.data( 'publish-text', selectedOption.data( 'publish-text' ) );
-					statusButton.needConfirm = true;
-				}
-
-				if ( 'future' === status ) {
-					snapshot.snapshotEditContainerDisplayed.set( true );
-					snapshot.snapshotExpandButton.show();
-					if ( snapshot.isFutureDate() ) {
-						snapshot.countdown.show();
-						snapshot.updateSnapshot( status );
-					}
-				} else {
-					snapshot.updateSnapshot( status );
-					snapshot.snapshotEditContainerDisplayed.set( false );
-					snapshot.countdown.hide();
+				toggleEditLinkControl: function() {
+					this.active.set( _.contains( [ 'draft', 'future', 'pending' ], api.state( 'changesetStatus' ).get() ) );
```

Comment:
I think the condition here should be opposite. Namely, it should be deactivated if the current status is `''` or `'auto-draft'`. This would be more future-proof if other plugins add their own statuses.